src/userobjects/OpenMCDomainFilterEditor.C

```diff
@@ -0,0 +1,204 @@
+  params.addParam<bool>("create_filter", false, "Whether to create the tally if it doesn't exist");
+  params.addRequiredParam<int32_t>("filter_id", "The ID of the filter to modify");
+  params.addParam<std::string>("filter_type", "", "The type of filter to create");
+  params.addRequiredParam<std::vector<std::string>>("bins", "The bins to modify in the filter");
+  params.declareControllable("bins");
+  params.addClassDescription("A UserObject for creating and managing OpenMC domain tally filters");
+  return params;
+}
+
+OpenMCDomainFilterEditor::OpenMCDomainFilterEditor(const InputParameters & parameters)
+  : OpenMCUserObject(parameters),
+    _filter_id(getParam<int32_t>("filter_id")),
+    _filter_type(getParam<std::string>("filter_type"))
+{
+}
+
+void
+OpenMCDomainFilterEditor::initialize()
+{
+  bool create_filter = getParam<bool>("create_filter");
+  bool filter_exists = this->filter_exists();
+
+  // check upon construction that the filter exists if we are not creating it
+  if (!create_filter && !filter_exists)
+  {
+    mooseError(long_name() + ": Filter " + std::to_string(_filter_id) +
+               " does not exist in the OpenMC XML model and create_filter is false");
+  }
+
+  // if create_filter is set to true, but the filter already exists, display a warning
+  if (create_filter && filter_exists)
+  {
+    mooseWarning(long_name() + ": Filter " + std::to_string(_filter_id) +
+                 " already exists in the OpenMC XML model");
+  }
+
+  // if this UO is to create the filter, but no filter type was specified, report an error
+  if (create_filter && !filter_exists && _filter_type.empty())
+  {
+    mooseError(long_name() + ": filter_type must be specified when create_filter = true");
+  }
+
+  // if the filter exists and the filter type was not specified, accept the existing filter type
+  if (filter_exists && _filter_type.empty())
+  {
+    // if an existing filter is being used, ensure that the filter type (if specified)
+    // matches the existing filter type
+    int32_t filter_index = openmc::model::filter_map.at(_filter_id);
+    _filter_type = openmc::model::tally_filters[filter_index]->type_str();
+    mooseWarning(
+        long_name() + ": Filter " + std::to_string(_filter_id) +
+        " already exists in the OpenMC model. Using existing filter type: " + _filter_type);
+  }
+
+  // at this point the filter type should be set, check that it is valid
+  if (_allowed_types.count(_filter_type) == 0)
+  {
+    std::string msg =
+        long_name() + ": Invalid filter type: " + _filter_type + ". Allowed types are: ";
+    for (const auto & type : _allowed_types)
+      msg += "\"" + type + "\"";
+    mooseError(msg);
+  }
+
+  // if the filter doesn't exist at this point and no other errors have been raised,
+  // create the filter
+  if (!filter_exists)
+  {
+    openmc_problem()->_console << long_name() << ": Creating Filter " << _filter_id << std::endl;
```

Comment:
why does this use `openmc_problem()->_console ` instead of just `_console`? I think they are the same?

Reply:
Didn't know that `_console` was available on the user objects. I'll use that instead!